pkg/transforms/mqttsecret.go

```diff
@@ -208,8 +212,27 @@ func (sender *MQTTSecretSender) MQTTSend(ctx interfaces.AppFunctionContext, data
 		sender.setRetryData(ctx, exportData)
 		return false, token.Error()
 	}
+	// capture the size for metrics
+	exportDataBytes := len(exportData)
+	if sender.MqttSizeMetrics == nil {
+		var err error
+		ctx.LoggingClient().Debugf("Initializing metric %s.", internal.MqttExportSizeName)
+		sender.MqttSizeMetrics = gometrics.NewHistogram(gometrics.NewUniformSample(internal.MetricsReservoirSize))
+		metricsManger := ctx.MetricsManager()
+		if metricsManger != nil {
+			// TODO: EdgeX 3.0 append topic to export size name
```

Comment:
We can do this now since this uses pointers for the receiver functions. i.e. we have a separate instance for each MQTTSecretSender. May want to use `host:port/topic` as the tag value.